unblob/extractor.py

```diff
@@ -36,14 +37,52 @@ def carve_chunk_to_file(carve_path: Path, file: io.BufferedIOBase, chunk: Chunk)
             f.write(data)
 
 
-def fix_permissions(outdir: Path):
+def fix_permission(path: Path):
+    if path.is_file():
+        path.chmod(0o644)
+    elif path.is_dir():
+        path.chmod(0o775)
+
+
+def is_safe_path(basedir: Path, path: Path) -> bool:
+    try:
+        basedir.joinpath(path).resolve().relative_to(basedir.resolve())
+    except ValueError:
+        return False
+    return True
+
+
+def fix_symlink(path: Path, outdir: Path) -> Path:
```

Comment:
do we want to fix infinite symlinks as well, eg:
```
marci@indigo:/tmp/sym$ ls -lR 
.:
total 4
drwxrwxr-x 2 marci marci 4096 febr  15 12:34 a

./a:
total 0
lrwxrwxrwx 1 marci marci 1 febr  15 12:34 a -> .
```

Reply:
Added a protection against recursive links (`is_recursive_link`) and wrote multiple integration tests to check both recursive links and link chains (`test_fix_symlink_chain`, `test_fix_symlink_chain_traversal`, `test_fix_recursive_symlink`).